homeassistant/components/media_source/local_source.py

```diff
@@ -71,17 +77,41 @@ async def async_browse_media(
     ) -> BrowseMediaSource:
         """Return media."""
         try:
-            source_dir_id, location = async_parse_identifier(item)
+            source_dir_id, location = self.async_parse_identifier(item)
         except Unresolvable as err:
             raise BrowseError(str(err)) from err
 
+        # If only one media dir is configured, use that as the local media root
+        if source_dir_id == "" and len(self.hass.config.media_dirs) == 1:
+            source_dir_id = list(self.hass.config.media_dirs)[0]
+
+        if source_dir_id == "":
+            base = BrowseMediaSource(
+                domain=DOMAIN,
+                identifier=None,
+                media_class=MEDIA_CLASS_DIRECTORY,
+                media_content_type=None,
+                title=self.name,
+                can_play=False,
+                can_expand=True,
+                children_media_class=MEDIA_CLASS_DIRECTORY,
+            )
+
+            base.children = [
+                await self.hass.async_add_executor_job(
```

Comment:
An optimization would be to make a function that calculates all the children and returns them and then just schedule that function on the thread pool.

Reply:
Yes. We should not await the threadpool in the event loop (passing around event loop and thread pool is "expensive") and instead just execute that whole loop in a thread.

Reply:
Done.